java/core/src/main/java/sleeper/core/record/process/status/ProcessRun.java

```diff
@@ -19,17 +19,21 @@
 import sleeper.core.record.process.RecordsProcessedSummary;
 
 import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
 
 public class ProcessRun {
     private final String taskId;
     private final ProcessRunStartedUpdate startedStatus;
     private final ProcessFinishedStatus finishedStatus;
+    private final List<ProcessStatusUpdate> statusUpdates;
 
     private ProcessRun(Builder builder) {
         taskId = Objects.requireNonNull(builder.taskId, "taskId must not be null");
-        startedStatus = Objects.requireNonNull(builder.startedStatus, "startedStatus must not be null");
+        startedStatus = builder.startedStatus;
         finishedStatus = builder.finishedStatus;
+        statusUpdates = Objects.requireNonNull(builder.statusUpdates, "statusUpdates must not be null");
```

Comment:
Sorry, it looks like this null check isn't needed now. We can just set it to be the value from the builder, maybe make it unmodifiable. It's a little confusing because everywhere else we have a list you can just set the whole thing in the builder.